cmd/kubeadm/app/master/selfhosted.go

```diff
@@ -69,9 +87,271 @@ func CreateSelfHostedControlPlane(cfg *kubeadmapi.MasterConfiguration, client *c
 		return err
 	}
 
+	if err := launchSelfHostedProxy(cfg, client); err != nil {
+		return err
+	}
+
+	if err := launchEtcdOperator(cfg, client); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getEtcdTPRClient() (*rest.RESTClient, error) {
+	kubeConfigPath := path.Join(kubeadmapi.GlobalEnvParams.KubernetesDir, kubeadmconstants.AdminKubeConfigFileName)
+	config, err := clientcmd.BuildConfigFromFlags("", kubeConfigPath)
+	if err != nil {
+		return nil, err
+	}
+
+	config.GroupVersion = &schema.GroupVersion{
+		Group:   spec.TPRGroup,
+		Version: spec.TPRVersion,
+	}
+	config.APIPath = "/apis"
+	config.ContentType = runtime.ContentTypeJSON
+	config.NegotiatedSerializer = serializer.DirectCodecFactory{CodecFactory: api.Codecs}
+
+	restcli, err := rest.RESTClientFor(config)
+	if err != nil {
+		return nil, err
+	}
+	return restcli, nil
+}
+
+func getBootEtcdPodIP(kubecli *clientset.Clientset) (string, error) {
+	var ip string
+	err := wait.Poll(5*time.Second, 60*time.Second, func() (bool, error) {
+		podList, err := kubecli.CoreV1().Pods(api.NamespaceSystem).List(metav1.ListOptions{
+			LabelSelector: "component=boot-etcd",
+		})
+		if err != nil {
+			fmt.Printf("failed to list 'boot-etcd' pod: %v\n", err)
+			return false, err
```

Comment:
I would expect that it's fine to not error out here but pass through the error if wait.Poll times out.